lib/carrierwave_data_uri/mount.rb

```diff
@@ -8,10 +8,15 @@ def mount_uploader(column, uploader=nil, options={}, &block)
         super
 
         class_eval <<-RUBY, __FILE__, __LINE__+1
+          attr_accessor :#{column}_data_uri
           attr_accessor :#{column}_data_filename
 
           def #{column}_data_uri=(data)
-            self.#{column} = Parser.new(data).to_file original_filename: self.#{column}_data_filename
+            original_filename = self.#{column}_data_filename.presence
+
+            if data.present?
```

Comment:
@kntmrkm i am thinking to add something like silent mode. If you enable this mode than parser won't raise an exception on bad data. So stay tuned